corehq/apps/api/resources/v0_5.py

```diff
@@ -1380,17 +1380,38 @@ def get_datasource_data(request, config_id, domain):
     return JsonResponse(data)
 
 
-@require_GET
-@api_auth()
-@requires_privilege_with_fallback(privileges.API_ACCESS)
-@api_throttle
-def get_cca_user_roles(request, domain):
-    if not toggles.SUPERSET_ANALYTICS.enabled(domain):
-        return HttpResponseForbidden(
-            _("This domain requires the superset-analytics flag to access this endpoint.")
-        )
+class CommCareAnalyticsAccessResource(v0_1.UserResource):
 
-    try:
-        return JsonResponse(get_commcare_analytics_access_for_user_domain(request.couch_user, domain))
-    except BadRequest as e:
-        return JsonResponse({'error': str(e)}, status=400)
+    roles = fields.ListField()
+    permissions = fields.DictField()
+
+    class Meta(v0_1.UserResource.Meta):
+        authentication = LoginAndDomainAuthentication()
+        object_class = CouchUser
+        resource_name = 'analytics-roles'
+
+    def dehydrate_role(self, bundle):
+        cca_access = get_commcare_analytics_access_for_user_domain(bundle.obj, bundle.request.domain)
+        return cca_access['roles']
+
+    def dehydrate_permissions(self, bundle):
+        cca_access = get_commcare_analytics_access_for_user_domain(bundle.obj, bundle.request.domain)
+        return cca_access['permissions']
+
+    def obj_get_list(self, bundle, **kwargs):
+        domain = kwargs['domain']
+        if not toggles.SUPERSET_ANALYTICS.enabled(domain):
+            raise ImmediateHttpResponse(
+                HttpForbidden(gettext_noop(
+                    "This domain requires the superset-analytics flag to access this endpoint."
```

Comment:
:heavy_plus_sign:

Reply:
@mkangia I decided to leave it as is. [Here's an example](https://github.com/dimagi/commcare-hq/blob/e8869e163f27278e0663d1fea90a6e426884eb5b/corehq/apps/api/resources/v0_5.py#L1059-L1063) where we do indeed give the user some info as part of the forbidden response through the api. In fact, I can only find 2 examples in the whole codebase for returning forbidden on an API resource, and both give some info. Maybe the question is what info to give?

If there's strong feelings otherwise I'm happy to update.

Reply:
Hey @Charl1996 

The reason for the change request here is that feature flags are not visible to the end users and is most probably unknown to most non-admin users. So, as a practice any information about feature flag is only visible to domain admins and superusers.
So, the change request here is simply in line with how HQ otherwise responds to requests that are behind feature flags, for domains/users that don't have access.
Here is a [link](https://github.com/dimagi/commcare-hq/blob/4a1dd045a92f3d819d6fedbcbcf315ad6bd9e130/corehq/toggles/__init__.py#L258) to the view decorator for feature flags. I wonder if you could directly use the decorator instead so you don't need to write this code.


As for the link you shared is for when a user does not have necessary permissions to access the link, so they see an error message that talks about permission. Most users are aware of roles and permissions, so that message seems reasonable. Noting that the message does not mention what permission specifically is missing too.

Reply:
> So, as a practice any information about feature flag is only visible to domain admins and superusers.

Ah, ok, that settles it then. Will remove the message.